src/evd/evd.cpp

```diff
@@ -558,141 +578,160 @@ int evd_process(evdOptions *opts)
+                if (!valid) continue;   //Skip rest of computation
 
-                //Check if coherence is positive semi-definite
-                int status = workers[threadnum].smallestEigen(Amp2.slice_memptr(threadnum), false);
-                //std::cout << "CORR Eig: "<< status << " " << workers[threadnum].eigval[0] << "\n";
+                //Perform Hadamard product
+                Amp2.slice(threadnum) %= Covar.slice(threadnum);
 
-                if (status !=0) 
-                {
-                    //std::cout << "Coher eig failed \n";
-                    corr[pp] = -3;
-                    valid = false;
-                }
-                else
+                //Obtain the smallest eigen vector
                 {
-                    if (workers[threadnum].eigval[0] < 1.0e-6)
+                    int status = workers[threadnum].smallestEigen( Amp2.slice_memptr(threadnum), true);
+                    if (status != 0)
                     {
-                        //std::cout << "Coher Eig: " << workers[threadnum].eigval[0] << "\n";
-                        corr[pp] = -4;
+                        corr[pp] = -6;
                         valid = false;
                     }
+                    else
+                    {
+                        if (workers[threadnum].eigval[0] < 1.0e-6)
+                        {
+                            corr[pp] = -7;
+                            valid = false;
+                        }
+                    }
                 }
-            }
-            if (!valid) continue;   //Skip rest of computation
-
-
-
-            //MLE stuff
-            //We have access to eigenvector as well as Covar here
-            //This is where the iterative MLE solver goes
-            //MLE solver should put solution back into evddata
+                if (!valid) continue;   //Skip updating evddata
 
+            } //End of MLE
+            else
             {
-                //copy coherence to Amp2
-                Amp2.slice(threadnum) = Amp1.slice(threadnum);
-
-                //Perform inverse operation
-                int status = workers[threadnum].positiveDefiniteInverse(Amp2.slice_memptr(threadnum));
-                if (status != 0)
+                //This is for SBAS / EVD - which both use the same concept
+                //Largest eigen vector of the correlation matrix
+               
+                //If SBAS, apply bandwidth limits
+                if (method.compare("SBAS")==0)
                 {
-                    std::cout << "Inverse failed \n";
-                    corr[pp] = -5;
-                    valid = false;
+                    
+                    //Blank out covariance entries outside the bandwidth
+                    for (int ti=0; ti <nbands; ti++)
+                    {
+                        for(int tj=ti+BW+1; tj< nbands; tj++)
+                        {
+                            Covar.at(tj,ti,threadnum) = 0.;
```

Comment:
Does this assignment need to be a std::complex value?

Reply:
Armadillo seems to take care of it - we had assignments of constant ones. Updated all to use std::complex just in case.